### PR TITLE
Enforce head style

### PR DESCRIPTION
**Pull request by richsalz**
This adds some more style rules to section/sub-section headings.
It is based on #9613, so only look at the top commit right now, or wait until that one is merged and I will rebase.

### CODE REVIEW

File: doc/man3/OSSL_trace_set_channel.pod [hunk 1]
```diff
@@ -214,8 +214,6 @@ the following:
      BIO_dump(trc_out, bar, sizeof(bar));
  } OSSL_TRACE_END(TLS);
 
-=head1 Simple example
-
```
Comment (levitte) [inline doc/man3/OSSL_trace_set_channel.pod]: It should remain as a head2

Comment (richsalz) [reply doc/man3/OSSL_trace_set_channel.pod]: okay, pushed.

(For what it's worth, I think we split things up with head2 way too often but that's just a side-note).

Comment (levitte) [reply doc/man3/OSSL_trace_set_channel.pod]: I disagree with you. The sub-sections makes it look less like walls of text.

Comment (richsalz) [reply doc/man3/OSSL_trace_set_channel.pod]: This isn't Moby Dick or Finnegan's Wake; it's a manpage and there's under 82,000 lines split over nearly 600 files.  Shrug.

File: doc/man3/EVP_EncryptInit.pod [hunk 2]
```diff
@@ -485,7 +485,7 @@ the length of the tag (with the C<tag> parameter set to NULL) when encrypting.
 The tag length is often referred to as B<M>. If not set a default value is
 used (12 for AES). When decrypting, the tag needs to be set before passing
 in data to be decrypted, but as in GCM and OCB mode, it can be set after
-passing additional authenticated data (see L<AEAD Interface>).
+passing additional authenticated data (see L<AEAD INTERFACE>).
```
Comment (levitte) [inline doc/man3/EVP_EncryptInit.pod]: Oh my...  would you mind correcting that link while you're at it?  It should be `L</AEAD INTERFACE>`.  Here's the partial explanation from `man perlpod`:

```
       "L<name>" -- a hyperlink
           There are various syntaxes, listed below.  In the syntaxes given,
           "text", "name", and "section" cannot contain the characters '/' and
           '|'; and any '<' or '>' should be matched.

           •   "L<name>"

               Link to a Perl manual page (e.g., "L<Net::Ping>").  Note that
               "name" should not contain spaces.  This syntax is also
               occasionally used for references to Unix man pages, as in
               "L<crontab(5)>".

           •   "L<name/"sec">" or "L<name/sec>"

               Link to a section in other manual page.  E.g., "L<perlsyn/"For
               Loops">"

           •   "L</"sec">" or "L</sec>"

               Link to a section in this manual page.  E.g., "L</"Object
               Methods">"
```

Comment (richsalz) [reply doc/man3/EVP_EncryptInit.pod]: fixup commit pushed.

**Comment by levitte** [comment]
You could do the squashing you want and force-push at this point. At your option, of course

**Comment by richsalz** [comment]
I prefer not to squash/push while review is ongoing because sometimes reviewers get confused.

BTW, if you want to backport this and it doesn't pick cleanly, I can create a backport PR.

File: doc/man1/cms.pod [hunk 3]
```diff
@@ -560,7 +560,7 @@ The B<-debug_decrypt> option can be used to disable the MMA attack protection
 and return an error if no recipient can be found: this option should be used
 with caution. For a fuller description see L<CMS_decrypt(3)>).
 
-=head1 CAdES Basic Electronic Signature (CAdES-BES)
+=head1 CADES BASIC ELECTRONIC SIGNATURE (CADES-BES)
```
Comment (InfoHunter) [inline doc/man1/cms.pod]: I am not sure if the 'd' in 'CADES' should be also uppercased. 'CAdES' sounds like a proper noun from my limited English knowledges...

Comment (richsalz) [reply doc/man1/cms.pod]: I know, I flip-flopped and went for consistency.

Comment (levitte) [reply doc/man1/cms.pod]: I believe the man-pages get head1s uppercased anyway. Maybe not html, though....

Comment (InfoHunter) [reply doc/man1/cms.pod]: Okay

**Comment by InfoHunter** [comment]
Do we consider to port this back to 1.1.1?

**Comment by richsalz** [comment]
I think it should be backported, yes.  If it doesn't pick cleanly, I will create a new PR.

**Comment by paulidale** [comment]
Merged!

**Pull request closed without merging**